### DISCUSSION TITLE
Export Users' Details (Including Passwords) from Keycloak

### DISCUSSION THREAD

**Discussion by divya-sk** [keycloak/keycloak#28459] [Q&A]
How to export users' details (Including Passwords) from Keycloak from one env to another without distrubing there passwords or other details

**Comment by codespearhead** [comment]
Are you migrating away from Keycloak to another Identity Provider, there might already be a migration guide available in their website.

Either way, you can just use the [Keycloak CLI](https://www.keycloak.org/server/importExport), which currently exports users with hashed passwords with [pbkdf2-sha512](https://www.keycloak.org/docs/latest/server_admin/#hashing-iterations) by default.

```bash
mkdir backup &&
./bin/kc.sh export --dir backup
```
